examples/google-tables/pack.ts

```diff
@@ -0,0 +1,184 @@
+    let tables = await getTables(context);
+    return tables.map(table => {
+      return {
+        display: table.displayName,
+        value: getTableUrl(table.name),
+      };
+    });
+  },
+  getName: async function (context) {
+    let tableUrl = context.sync!.dynamicUrl!;
+    let table = await getTable(context, tableUrl);
+    return table.displayName;
+  },
+  getSchema: async function (context) {
+    let tableUrl = context.sync!.dynamicUrl!;
+    let table = await getTable(context, tableUrl);
+    let schema: coda.GenericObjectSchema = {
+      ...BaseRowSchema,
+      properties: {
+        ...BaseRowSchema.properties,
+      },
+    };
+    for (let column of table.columns) {
+      let property = getPropertySchema(column, table, context);
+      if (!property) {
+        continue;
+      }
+      schema.properties[column.name] = property;
+      schema.featuredProperties!.push(column.name);
+    }
+    return schema;
+  },
+  getDisplayUrl: async function (context) {
+    let tableUrl = context.sync!.dynamicUrl!;
+    let tableId = tableUrl.split("/").pop() as string;
+    return coda.joinUrl("https://tables.area120.google.com/table", tableId);
+  },
+  // Get the options for select list columns.
+  propertyOptions: async function (context) {
+    let tableUrl = context.sync!.dynamicUrl!;
+    let columnId = context.propertyName;
+    let table = await getTable(context, tableUrl);
+    let column = table.columns.find(column => column.id === columnId);
+    if (!column) {
+      throw new Error(`Cannot find column: ${columnId}`);
+    }
+    if (column.labels) {
+      return column.labels.map(label => label.name);
+    }
+    return undefined;
+  },
+  formula: {
+    name: "SyncTable",
+    description: "Syncs the data.",
+    parameters: [],
+    execute: async function (args, context) {
+      let {pageToken, rowNumber = 1} = (context.sync.continuation ??
+        {}) as RowsContinuation;
+      let tableUrl = context.sync.dynamicUrl!;
+      let table = await getTable(context, tableUrl);
+      let {rows, nextPageToken} = await getRows(context, tableUrl, pageToken);
+      let formattedRows = rows.map((row: any) => {
+        return formatRowForSchema(row, table, context, String(rowNumber++));
+      });
+      let continuation: RowsContinuation | undefined;
+      if (nextPageToken) {
+        continuation = {
+          pageToken: nextPageToken,
+          rowNumber: rowNumber,
+        };
+      }
+      return {
+        result: formattedRows,
+        continuation: continuation,
+      };
+    },
+    maxUpdateBatchSize: 10,
+    executeUpdate: async function (args, updates, context) {
+      let tableUrl = context.sync.dynamicUrl!;
+      let table = await getTable(context, tableUrl);
+
+      // Create an async job for each update.
+      let jobs = updates.map(async update => {
+        // Convert the row back to an API format.
+        let row = formatRowForApi(update.newValue, table, context);
```

Comment:
Do you need a try/catch around this to report per-row errors correctly?

Reply:
I don't think so. I use `Promise.allSettled` below, which doesn't throw when a row fails but instead returns the error.